sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -337,9 +346,23 @@ def _delete(self, **kwargs):
         :returns: A dictionary containing information about the deleted path.
         :rtype: Dict[str, Any]
         """
-        options = self._delete_path_options(**kwargs)
+        # Perform paginated delete only if using OAuth and api version is 2023-08-03 or later
+        # The pagination is only for ACL checks, the final requests remains the atomic delete operation
+        paginated = None
+        if (compare_api_versions(self.api_version, '2023-08-03') >= 0 and
+            hasattr(self.credential, 'get_token')):
+            paginated = True
+
+        options = self._delete_path_options(paginated, **kwargs)
         try:
-            return self._client.path.delete(**options)
+            response_headers = self._client.path.delete(**options)
+            # Loop until continuation token is None for paginated delete
+            while response_headers['continuation']:
+                response_headers = self._client.path.delete(
+                    continuation=response_headers['continuation'],
+                    **options)
+
+            return response_headers
```

Comment:
This will be returning an empty `continuation` header right?
Is there ever a scenario where customers might want this value and manually continue to pass it in?
This feels like the kind of function where we really should support some kind of client-level timeout where I can bail out of this loop if it starts taking longer than I expected.... what could be the worst case blocking time here?

Reply:
Yes, this will return an empty `continuation` as it does today. I'm not sure the worst-case blocking time here but it could theoretically get pretty large if the directory is huge. Currently customers that call this API today would also be blocked on the single call as the service would not paginate and I'm not sure we get many/any complaints on that. I also don't anticipate many customers will actually run into this scenario as you need to be using OAuth with file level ACLs rather than RBAC permissions to the Storage account for this to paginate. I don't think this is super common workflow, at least for REST API usage (maybe SMB, NFS, etc.). One other thing to note is that the continuation token only lasts for 4 minutes so saving it and passing it in later doesn't make too much sense as it would likely be expired by the time you try and start again.

So ultimately my opinion is that we don't need to add any special stuff here unless we get feedback from customers in this area. Timeouts are probably things we can always add later if needed.

Reply:
The difference is that previously with the single call implementation, a customer could pass in `timeout` to specify the service-side timeout and that would have prevented a blocking operation. Now that same customer will find their timeout no longer works because it will be applied separately to multiple operations and not cumulatively.

Reply:
As discussed offline, we will go ahead and proceed with this as-is for preview but circle back on timeouts before GA.